test/app-luatest/fiber_slice_test.lua

```diff
@@ -186,11 +186,11 @@ g.test_compat = function()
         local compat = require('compat')
         local fiber = require('fiber')
         t.assert_equals(compat.fiber_slice_default.current, 'default')
-        t.assert_equals(compat.fiber_slice_default.default, 'old')
-        t.assert_equals(fiber.self():info().max_slice, nil)
-        compat.fiber_slice_default = 'new'
+        t.assert_equals(compat.fiber_slice_default.default, 'new')
         t.assert_equals(fiber.self():info().max_slice, {warn = 0.5, err = 1.0})
         compat.fiber_slice_default = 'old'
         t.assert_equals(fiber.self():info().max_slice, nil)
+        compat.fiber_slice_default = 'new'
```

Comment:
you test defaults, above you check that default value is `new`, so why do you need to set `compat.fiber_slice_default` to `new` explicitly?

Reply:
I think, it is good to check compat option in the following way: 
1) test default (`new` here)
2) test opposite of default (`old` here)
3) test the same as default but explicitly set (`new` here)
 
This way, we can check that the compat action function can switch for both `old -> new` and `new -> old`. Note, that when current value is the same as the one being set, the action function is not called.

Reply:
> test default (new here)
> test opposite of default (old here)
> test the same as default but explicitly set (new here)

I like approach, and it would be good to test each step in a separate testcase in the future.